lib/karto_sdk/include/karto_sdk/Karto.h

```diff
@@ -5658,27 +5658,21 @@ class LocalizedRangeScan : public LaserRangeScan
       kt_int32u beamNum = 0;
       for (kt_int32u i = 0; i < pLaserRangeFinder->GetNumberOfRangeReadings(); i++, beamNum++) {
         kt_double rangeReading = GetRangeReadings()[i];
-        if (!math::InRange(rangeReading, pLaserRangeFinder->GetMinimumRange(), rangeThreshold)) {
-          kt_double angle = scanPose.GetHeading() + minimumAngle + beamNum * angularResolution;
-
-          Vector2<kt_double> point;
-          point.SetX(scanPose.GetX() + (rangeReading * cos(angle)));
-          point.SetY(scanPose.GetY() + (rangeReading * sin(angle)));
-
-          m_UnfilteredPointReadings.push_back(point);
+        if (!math::InRange(rangeReading, pLaserRangeFinder->GetMinimumRange(),
+                           pLaserRangeFinder->GetMaximumRange()))
           continue;
-        }
 
         kt_double angle = scanPose.GetHeading() + minimumAngle + beamNum * angularResolution;
 
         Vector2<kt_double> point;
         point.SetX(scanPose.GetX() + (rangeReading * cos(angle)));
         point.SetY(scanPose.GetY() + (rangeReading * sin(angle)));
 
-        m_PointReadings.push_back(point);
+        if (rangeReading < rangeThreshold) {
+            m_PointReadings.push_back(point);
```

Comment:
I believe that the occupancy grid generator needs the number of point readings to be the same as the number coming in since it assumes a regularized angle of each laser beam. If you drop points, then I think that will mess up the scan matching and/or the occupancy grid generation.

Reply:
Can you test if you remove some large number of scan points that the output map still looks correct & that scan matching still works properly?